torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -60,25 +54,16 @@ py::object PyRRef::localValue() {
       "Cannot call localValue() on a non-local reference. Call it on ",
       owner().name_);
 
-  if (rref_->isPyObj()) {
-    const py::object& value = jit::toPyObject(
-        std::dynamic_pointer_cast<OwnerRRef>(rref_)->getValue());
-    PythonRpcHandler::getInstance().handleException(value);
-    {
-      // acquiring GIL as the return statement construct a new py::object from
-      // a const reference.
-      pybind11::gil_scoped_acquire ag;
-      return value;
-    }
-  } else {
-    auto value = std::dynamic_pointer_cast<OwnerRRef>(rref_)->getValue();
-    {
-      // acquiring GIL as torch::jit::toPyObject creates new py::object without
-      // grabbing the GIL.
-      pybind11::gil_scoped_acquire ag;
-      return torch::jit::toPyObject(std::move(value));
-    }
+  py::object res;
+  auto value = std::dynamic_pointer_cast<OwnerRRef>(rref_)->getValue();
+  {
+    // acquiring GIL as torch::jit::toPyObject creates new py::object without
+    // grabbing the GIL.
+    pybind11::gil_scoped_acquire ag;
```

Comment:
(can be done in followup PR) do we need to use `PROFILE_GIL_SCOPED_ACQUIRE` to profile this?